ckanext/geocat/commands.py

```diff
@@ -21,37 +24,75 @@ def command(self):
             'search': self.searchCmd,
             'cql': self.cqlCmd,
             'dataset': self.datasetCmd,
+            'list': self.listCmd,
             'help': self.helpCmd,
         }
 
         try:
             cmd = self.args[0]
             options[cmd](*self.args[1:])
-        except KeyError:
+        except (KeyError, IndexError):
             self.helpCmd()
 
     def helpCmd(self):
         print self.__doc__
 
-    def cqlCmd(self, query=None):
+    def cqlCmd(self, query=None, csw_url=None):
         if (query is None):
             print "Argument 'query' must be set"
             self.helpCmd()
             sys.exit(1)
-        csw = md.CswHelper('http://www.geocat.ch/geonetwork/srv/eng/csw')
+        if csw_url is None:
+            csw_url = 'http://www.geocat.ch/geonetwork/srv/eng/csw'
```

Comment:
The same URL-Value is used several times. It is just a default, but maybe it is easier to read, if it is set once as a constant and re-used as a `default_url`.
Same with `default_cql` maybe?